File: beecrowd/python/estruturas-e-bibliotecas/ex1069.py
def extrair_diamantes(mina):
    total_diamantes = 0
    pilha = []
    for item in mina:
        if item == '<':
            pilha.append('<')
        elif item == '>':
            if len(pilha) > 0:
                pilha.pop()
                total_diamantes += 1
            else:
                continue
    return total_diamantes


num_entradas = int(input())
entradas = []

for i in range(num_entradas):
    entradas.append(input())
    
for entrada in entradas:
    print(extrair_diamantes(entrada))

File: beecrowd/c++/estruturas-e-bibliotecas/ex1069.cpp
#include <iostream>
#include <stack>

using namespace std;

int main(){
    int n;
    string mina;
    stack<char> pilha;

    cin>>n;
    cin.ignore();
    for(int i=0; i<n; i++){
        int qtd=0;
        getline(cin, mina);
        for (char c : mina){
            if (c=='<'){
                pilha.push('<');
            } else if(c=='>'){
                if(!pilha.empty()){
                    qtd+=1;
                    pilha.pop();
                }
            }
        }
        cout<<qtd<<endl;
        while(!pilha.empty()){
            pilha.pop();
        }
    }

    return 0;
}